### PR TITLE
add the search all post function

### PR DESCRIPTION
**Pull request by TomYYHong**

### CODE REVIEW

File: src/post/post.service.ts [hunk 1]
```diff
@@ -21,6 +21,19 @@ export class PostService {
     return createdPost;
   }
 
+  async fetchPosts() {
+    try{
+    // Use the model to find all records, limit the result to 50, and sort by date in descending order
+    return this._postModel
+      .find()
+      .limit(50)
+      .sort({ postedDate: -1 }) // -1 for descending order, 1 for ascending order
```
Comment (511234) [inline src/post/post.service.ts:29]: it's better to do sorting before limit(50)?

File: src/post/post.controller.ts [hunk 2]
```diff
@@ -19,6 +19,13 @@ export class PostController {
     return foundPost;
   }
 
+  //search ALL post related with one user id
+  @Get('/searchPosts/all')
```
Comment (511234) [inline src/post/post.controller.ts]: `@Get('/all')`

because `get` means to search posts

Comment (TomYYHong) [reply src/post/post.controller.ts]: solved

File: src/post/post.service.ts (hunk 1 above)
Comment (TomYYHong) [reply src/post/post.service.ts:29]: solved

**Review by 511234 — APPROVED** [review]

**Pull request merged**